### PR TITLE
Inline Representation: Sections by function, not class

### PR DESCRIPTION
**Pull request by kba**
#51

### CODE REVIEW

File: 1.2/spec.md [hunk 1]
```diff
@@ -551,73 +551,76 @@ Something that requires JPEG or PNG to be represented well
...
 
-### Hyphenation
+Hyphenation {#hyphenation}
+-----------
 
 Soft hyphens must be represented using the HTML `&shy;` entity.
 
 The HTML <a href="https://www.w3.org/TR/REC-html40/struct/dirlang.html#h-8.2.5">`&lrm;` and
 `&rlm;` entities</a> (indicating writing direction) must not be used; all
 writing direction changes must be indicated with tags.
```
Comment (amitdo) [inline 1.2/spec.md]: should be under 'Writing Direction' header

Comment (kba) [reply 1.2/spec.md]: Right, doesn't fit in hyphenation. Maybe move this to
https://kba.github.io/hocr-spec/1.2/#font-lang, replace "with tags" with  "`dir=` attribute" and reference https://kba.github.io/hocr-spec/1.2/#valdef-ocr-capabilities-ocrp_dir?

Comment (amitdo) [reply 1.2/spec.md]: +1

File: 1.2/index.html [hunk 2]
```diff
@@ -1525,19 +1525,13 @@ <h2 class="no-num no-toc no-ref" id="contents">Table of Contents</h2>
...
-        <li><a href="#non-default-spaces"><span class="secno">6.1.8</span> <span class="content">Non-default spaces</span></a>
-        <li><a href="#hyphenation"><span class="secno">6.1.9</span> <span class="content">Hyphenation</span></a>
-       </ol>
+      <li><a href="#unrecognized"><span class="secno">6.1</span> <span class="content">Unrecognized characters and words</span></a>
+      <li><a href="#dropcap"><span class="secno">6.2</span> <span class="content">Dropcap</span></a>
+      <li><a href="#formulas"><span class="secno">6.3</span> <span class="content">Mathematical and chemical formulas</span></a>
+      <li><a href="#sub-sup"><span class="secno">6.4</span> <span class="content">Superscript and Subscript</span></a>
+      <li><a href="#whitespace"><span class="secno">6.5</span> <span class="content">Whitespace</span></a>
+      <li><a href="#hyphenation"><span class="secno">6.6</span> <span class="content">Hyphenation</span></a>
+      <li><a href="#ruby"><span class="secno">6.7</span> <span class="content">Ruby characters</span></a>
```
Comment (zuphilip) [inline 1.2/index.html:1534]: You could also combine some of these sections under one new section, i.e.

**HTML entities and Unicode**
- Non-breaking spaces must be represented using the HTML `&nbsp;` entity.
- Different space widths should be indicated using HTML and `&ensp;`, `&emsp;`, `&thinsp;`, `&zwnj;`, `&zwj;`.
- Soft hyphens must be represented using the HTML `&shy;` entity.
- The HTML `&lrm;` and `&rlm;` entities (indicating writing direction) must not be used; all writing direction changes must be indicated with tags.
- Furigana and similar constructs must be represented using their correct Unicode encoding.

File: 1.2/spec.md [hunk 3]
```diff
@@ -551,73 +551,76 @@ Something that requires JPEG or PNG to be represented well
...
 
-### Non-breaking space
+Whitespace {#whitespace}
+----------
 
 Non-breaking spaces must be represented using the HTML `&nbsp;` entity.
 
-### Non-default spaces
-
 Different space widths should be indicated using HTML and `&ensp;`, `&emsp`,
```
Comment (zuphilip) [inline 1.2/spec.md]: Semicolon missing in `&emsp;`

File: 1.2/spec.md [hunk 4]
```diff
@@ -551,73 +551,76 @@ Something that requires JPEG or PNG to be represented well
...
+: <dfn element>ocr_math</dfn>
+:: A mathematical formula
+:: Must contain either a single <{img}> tag or [[MathML]] markup, or be present on one
+
+Superscript and Subscript {#sub-sup}
+-------------------------
 
 Superscripts and subscripts, when not in <{ocr_math}> or <{ocr_chem}> formulas,
-must be represented using the HTML `<sup>` and `<sub>` tags, even if special
+must be represented using the HTML <{sup}> and <{sub}> tags, even if special
```
Comment (zuphilip) [inline 1.2/spec.md:604]: These links are not working and I am not sure there is anything we can link to...

Comment (kba) [reply 1.2/spec.md:604]: Fixed, thanks https://www.w3.org/TR/html5/text-level-semantics.html#the-sub-and-sup-elements

File: 1.2/spec.md (hunk 3 above)
Comment (kba) [reply 1.2/spec.md]: Fixed.

File: 1.2/index.html (hunk 2 above)
Comment (kba) [reply 1.2/index.html:1534]: I think these aspects are worth their own section. e.g. for Whitespace: explain whether repeated whitespace is meaningful, if non-tabular aligned text should use tabs. For hyphenation, whether that's the only encoding (e.g. https://github.com/altoxml/schema/issues/41).

Comment (zuphilip) [reply 1.2/index.html:1534]: Hyphens are also mentioned in http://kba.github.io/hocr-spec/1.2/#hardbreak

Besides ruby also other special entities are mentioned in the article:

> For example, HTML and CSS provide
> support for representing fonts, styles, hyphenation,
> flexible spacing, justification, kashida (flexible Arabic
> characters), Urdu ligatures, Japanese ruby, mixed hor-
> izontal/vertical layout, inline changes in writing direc-
> tion, and many others.

However, I am also fine with more subsections.

Comment (kba) [reply 1.2/index.html:1534]: I've moved the paragraph there before and probably will again once I get to the fonts/language section :)

Comment (kba) [reply 1.2/index.html:1534]: https://github.com/kba/hocr-spec/pull/60

**Comment by amitdo** [comment]
`<sub>` and `<sup>` for html 4.01:
https://www.w3.org/TR/html401/struct/text.html#h-9.2.3

**Comment by kba** [comment]
I just find the HTML5 standard way better. I know we have

> all tags should be used for the intended purpose (and only for the intended purpose) as defined in the  [HTML40] spec.

in there, but we should rather change that than link to an old spec with bad examples:

``` html
      H<sub>2</sub>O
      E = mc<sup>2</sup>
      <SPAN lang="fr">M<sup>lle</sup> Dupont</SPAN>
```

The first two should not use sub/sup at all. None of the tags should be upper-case.

**Pull request merged**